### PR TITLE
BZ1747486 - adding hostPath volume doc

### PR DESCRIPTION
**Pull request by bobfuru**
[BZ 1747486](https://bugzilla.redhat.com/show_bug.cgi?id=1747486)
Added hostPath config example, based on upstream docs.
@sferich888 and @liangxia PTAL

### CODE REVIEW

**Comment by liangxia** [comment]
LGTM.
Need to CP to 4.1, too.

**Comment by bobfuru** [comment]
@sferich888 Bump, PTAL.

File: modules/persistent-storage-hostpath-mounting.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// * storage/persistent-storage/persistent-storage-hostpath.adoc
+
+[id="persistent-storage-hostpath-mounting_{context}"]
+= Mounting with hostPath
+
+{product-title} supports hostPath mounting for development and testing on a single-node cluster.
+
+In a production cluster, you would not use hostPath. Instead, a cluster administrator would provision a network resource, such as a GCE Persistent Disk volume, an NFS share, or an Amazon EBS volume.
+
+Cluster administrators can also use StorageClasses to set up dynamic provisioning.
```
Comment (sferich888) [inline modules/persistent-storage-hostpath-mounting.adoc]: Do we need to show how to set up the storage class for this?

Comment (bobfuru) [reply modules/persistent-storage-hostpath-mounting.adoc]: @sferich888 I'm investigating. hostPath dynamic provisioning is not supported, so my understanding is that storage class may not be needed at all.

**Comment by bobfuru** [comment]
@liangxia 
@huffmanca 
PTAL - I have refactored this assembly to give the user instructions for configuring persistent storage using hostPath. Confirmed with Dev that dynamic provisioning is not supported. Updates to this PR are intended to give the user instructions for defining the PV and PVC, as well as calling out that Pods must be run as privileged. Also following up with PM about support level.

File: modules/persistent-storage-hostpath-static-provisioning.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: task-pvc-volume
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: slow
```
Comment (liangxia) [inline modules/persistent-storage-hostpath-static-provisioning.adoc]: The PVC's storageClassName mismatch with PV's storageClassName.

**Comment by liangxia** [comment]
Other than the storageClassName mismatch, LGTM.

**Comment by chao007** [comment]
Can we add a pod file here?
For hostpath volume, we need to use privileged pod.

File: modules/persistent-storage-hostpath-static-provisioning.adoc (hunk 2 above)
Comment (bobfuru) [reply modules/persistent-storage-hostpath-static-provisioning.adoc]: Updated so they are both using `manual` now.

**Comment by bobfuru** [comment]
> Can we add a pod file here?
> For hostpath volume, we need to use privileged pod.

Yes, good suggestion. I have added `modules/persistent-storage-hostpath-pod.adoc` to this commit.

**Comment by bobfuru** [comment]
@liangxia or @chao007 
PTAL at revised version with added Pod file

@sferich888 
Can I also get an ack that this looks good to you?

File: modules/persistent-storage-hostpath-pod.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+//
+// * storage/persistent-storage/persistent-storage-hostpath.adoc
+
+[id="persistent-storage-hostpath-pod_{context}"]
+= Mounting the hostPath share in a privileged Pod
+
+After the PersistentVolumeClaim has been created, it can be used inside by an application. The following example demonstrates mounting this share inside of a Pod.
+
+.Prerequisites
+* A PersistentVolumeClaim exists that is mapped to the underlying hostPath share.
```
Comment (sferich888) [inline modules/persistent-storage-hostpath-pod.adoc]: Should we link to docs on creating this pvc?

File: modules/persistent-storage-hostpath-static-provisioning.adoc [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent-storage/persistent-storage-vsphere.adoc
+
+[id="vsphere-static-provisioning_{context}"]
+= Statically provisioning hostPath volumes
```
Comment (sferich888) [inline modules/persistent-storage-hostpath-static-provisioning.adoc]: Don't we list this in https://github.com/openshift/openshift-docs/pull/17603/files#diff-1c65246cb57e25d53ab01b23a5d211de

File: modules/persistent-storage-hostpath-pod.adoc (hunk 3 above)
Comment (sferich888) [reply modules/persistent-storage-hostpath-pod.adoc]: /lgtm after we consider this

**Comment by chao007** [comment]
> > Can we add a pod file here?
> > For hostpath volume, we need to use privileged pod.
> 
> Yes, good suggestion. I have added `modules/persistent-storage-hostpath-pod.adoc` to this commit.

Thanks.

File: modules/persistent-storage-hostpath-pod.adoc (hunk 3 above)
Comment (bobfuru) [reply modules/persistent-storage-hostpath-pod.adoc]: The steps for creating this PVC are actually described in the previous module (modules/persistent-storage-hostpath-static-provisioning.adoc). But unfortunately, our toolchain prevents the use of module-to-module xrefs. And because each 4.x module is written to serve as standalone content so that it may be reused anywhere else in our docs, we also can't refer to "the previous step." So, I am leaving it as is.

**Comment by bobfuru** [comment]
@huffmanca PTAL - acks received from QE and Support. Thanks!

File: modules/persistent-storage-hostpath-pod.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      name: hostpath-privileged
+  ...
+  securityContext: {}
+  volumes:
+    - name: hostpath-privileged
+      persistentVolumeClaim:
+        claimName: task-pvc-volume <4>
+----
+<1> The name of the Pod.
+<2> The Pod must run as privileged to grant access from mounted storage to data across Pods in the same node.
```
Comment (huffmanca) [inline modules/persistent-storage-hostpath-pod.adoc]: This sentence is a bit wordy. I think we can just say `The Pod must run as privileged to access the node's storage.`

**Review by huffmanca — COMMENTED** [review]
Only a couple of minor things. The rest looks good to me!

File: storage/persistent-storage/persistent-storage-hostpath.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+[id="persistent-storage-using-hostpath"]
+= Persistent storage using hostPath
+include::modules/common-attributes.adoc[]
+:context: persistent-storage-hostpath
+
+toc::[]
+
+A _hostPath_ volume in an {product-title} cluster mounts a file or directory from the host node’s filesystem into your Pod. Most Pods will not need a hostPath volume, but it does offer powerful protection should an application require it.
```
[the remark this answers is not in the document]
Comment (bobfuru) [reply storage/persistent-storage/persistent-storage-hostpath.adoc]: I see what you mean. Original wording was "powerful escape hatch" so was looking for something more translatable but "protection" is not the right choice. Changing it to say that "it does offer a quick option for testing should an application require it."
